brats/transformations.py

```diff
@@ -73,3 +76,87 @@ def hist_match(source: np.ndarray, template: np.ndarray) -> np.ndarray:
                                     template_values)
         result[positive] = interp_t_values[bin_idx]
         return result
+
+
+class NiftiOrderTransformation:
+    """
+    Changes dimensions order from nifti (H,W,D,C) to torch convention (C,D,W,H).
+    It can be applied to both ``torch.Tensor`` or ``numpy.ndarray``.
+    """
+
+    def __call__(self, img):
+        if type(img) not in [np.ndarray, torch.Tensor]:
+            raise ValueError("img should be either np.ndarray or torch.Tensor")
+        if len(img.shape) != 4:
+            raise ValueError("img should have 4 dimensions (H,W,D,C)")
+        if isinstance(img, torch.Tensor):
+            transformed = img.permute(3, 2, 0, 1)
+        if isinstance(img, np.ndarray):
+            transformed = np.moveaxis(img, [0, 1, 2, 3], [2, 3, 1, 0])
+        return transformed
+
+
+class AddChannelDimToMaskTransformation:
+    """
+    Adds additional channel dimension
+    It can be applied to both ``torch.Tensor`` or ``numpy.ndarray``.
+    """
+
+    def __call__(self, img):
+        if type(img) not in [np.ndarray, torch.Tensor]:
+            raise ValueError("img should be either np.ndarray or torch.Tensor")
+        if len(img.shape) != 3:
+            raise ValueError("img should have 3 dimensions (H,W,D)")
+        if isinstance(img, torch.Tensor):
+            transformed = torch.unsqueeze(img, 3)
+        if isinstance(img, np.ndarray):
+            transformed = np.expand_dims(img, 3)
+        return transformed
+
+
+class BinarizationTransformation:
+    """
+    Adds additional channel dimension
+    I can be applied to``numpy.ndarray``.
+    """
+
+    def __call__(self, img):
+        if type(img) not in [np.ndarray, torch.Tensor]:
+            raise ValueError("img should be either np.ndarray or torch.Tensor")
```

Comment:
All of lines where u check type sohuld raise `TypeError` not `ValueError`. Check whole file for this.

Reply:
Not relevant anymore